src/kernel/capinv.c

```diff
@@ -1449,6 +1477,19 @@ composite_syscall_slowpath(struct pt_regs *regs, int *thd_switch)
 				if (unlikely(ret < 0)) cos_throw(err, ret);
 
 				*thd_switch = 1;
+			} else if(tcap_expended(tcap_current(cos_info))) {
+				struct thread *thd_next;
```

Comment:
Since this logic is identical to the logic above (see 1425), it should be in a function